### ISSUE TITLE
Pyright evaluates both sides of a ternary even when it shouldn't.

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#4688]
**To Reproduce**
Pyright will evaluate both sides of a ternary even when a Boolean value is known at type-check time.

**Expected behaviour**
Only the side known to run at type check time should be evaluated. E.g in the code below type of `foo` should be `Literal[1]`

**Screenshots or Code**
```python
from typing import TYPE_CHECKING

foo = 1 if TYPE_CHECKING else 'a'
reveal_type(foo)  # Type of "foo" is "Literal[1, 'a']"
```

**VS Code extension or command-line**
pyright v1.1295

**Additional context**
Easy enough to work around by using
```py
if TYPE_CHECKING:
    foo = 1
else:
    foo = 'a'
```
But it seems weird to me the ternary doesn't work.

**Comment by erictraut** [comment]
This behavior is as designed. You should use an if/else statement if you want code to be not evaluated. Evaluation is skipped at the level of individual statements, not partial statements like individual operands.

**Comment by erictraut** [comment]
I'll investigate to see if there's a straightforward way to support this in the ternary expression evaluation code.

**Comment by erictraut** [comment]
I was able to add support for this with minimal added complexity and negligible performance overhead. It will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.296, which I just published. It will also be included in a future release of pylance.
